addons/stock/models/stock_move.py

```diff
@@ -523,14 +523,19 @@ def _set_lot_ids(self):
                     move_line.qty_done = 1
             move.write({'move_line_ids': move_lines_commands})
 
-    @api.depends('picking_type_id', 'date', 'priority')
+    @api.depends('picking_type_id', 'picking_type_id.reservation_method', 'picking_type_id.reservation_days_before', 'picking_type_id.reservation_days_before_priority', 'date', 'state', 'priority')
     def _compute_reservation_date(self):
         for move in self:
-            if move.picking_type_id.reservation_method == 'by_date' and move.state in ['draft', 'confirmed', 'waiting', 'partially_available']:
-                days = move.picking_type_id.reservation_days_before
-                if move.priority == '1':
-                    days = move.picking_type_id.reservation_days_before_priority
-                move.reservation_date = fields.Date.to_date(move.date) - timedelta(days=days)
+            reservation_date = False
+            if move.state in ['confirmed', 'partially_available']:
+                if move.picking_type_id.reservation_method == 'at_confirm':
+                    reservation_date = fields.Date.today()
+                elif move.picking_type_id.reservation_method == 'by_date':
+                    days = move.picking_type_id.reservation_days_before
+                    if move.priority == '1':
+                        days = move.picking_type_id.reservation_days_before_priority
+                    reservation_date = fields.Date.to_date(move.date) - timedelta(days=days)
+            move.reservation_date = reservation_date
```

Comment:
I will keep the original behavior, don't set false.
To keep the value (when state is done), or if a custom dev write directly reservation_date.

Reply:
I think the whole idea of setting values to `False` if a reservation date is not needed was probably to have only those which needs reservation in the pool, but maybe I am wrong.

Custom devs should IMO inherit the method and can do their own logic...I do oppose hard the opinion that a computed field should manipulate values from the outside of the method itself...this is just wrong or do I miss something?